preprocess/cache_main.cc

```diff
@@ -102,22 +163,47 @@ void Output(util::UnboundedSingleQueue<QueueEntry> &queue, util::scoped_fd &proc
   }
 }
 
+// Parse arguments using boost::program_options
+void ParseArgs(int argc, char *argv[], Options &out) {
+  namespace po = boost::program_options;
+  po::options_description desc("Acts as a cache around another program processing one line in, one line out from stdin to stdout.");
+  desc.add_options()
+    ("key,k", po::value(&out.key)->default_value("-1"), "Column(s) key to use as the deduplication string")
+    ("field_separator,t", po::value(&out.field_separator)->default_value("\t"), "use a field separator instead of tab");
+  po::variables_map vm;
+  po::store(po::parse_command_line(argc, argv, desc), vm);
+  po::notify(vm);
+}
+
 int main(int argc, char *argv[]) {
   // The underlying program can buffer up to kQueueLength - kFlushRate lines.  If it goes above that, deadlock waiting for queue.
   const std::size_t kFlushRate = 4096;
-  if (argc < 2) {
-    std::cerr << "Acts as a cache around another program processing one line in, one line out from stdin to stdout.\n"
-      "Usage: " << argv[0] << " slow arguments_to_slow\n";
-    return 1;
+  
+  // Take into account the number of arguments given to `cache` to delete them from the argv provided to Launch function
+  Options opt;
+  int skip_args = 1;
+  std::string s1(argv[1]);
```

Comment:
We're converting a `char *` to a `std::string` just so we can compare it?  `strcmp`.  This is messy but I get why you are doing it.